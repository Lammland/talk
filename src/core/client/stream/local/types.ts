import { GQLCOMMENT_SORT_RL } from "coral-framework/schema";

export type COMMENTS_TAB =
  | "NONE"
  | "ALL_COMMENTS"
  | "FEATURED_COMMENTS"
  | "QUESTIONS"
  | "REVIEWS"
  | "UNANSWERED_COMMENTS"
  | "%future added value";

export type ACTIVE_TAB =
  | "COMMENTS"
  | "PROFILE"
  | "DISCUSSIONS"
  | "%future added value";

export type COMMENT_SORT = GQLCOMMENT_SORT_RL;

export type AUTH_VIEW =
  | "FORGOT_PASSWORD"
  | "SIGN_IN"
  | "SIGN_UP"
  | "%future added value";
