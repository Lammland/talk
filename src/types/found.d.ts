import { ElementsRenderer, LinkPropsWithFunctionChild, Link } from "found";

declare module "found" {
  // TODO: (cvle) remove when merged https://github.com/4Catalyzer/found/pull/780
  const ElementsRenderer: ElementsRenderer;

  const Link: any;
}

declare module "found/ResolverUtils";
