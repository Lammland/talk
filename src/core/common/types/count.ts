export interface CountJSONPData {
  ref: string;
  html: string;
  count: number;
  seenCount?: number | null;
  id?: string | null;
}

export interface PreviousCountData {
  count: number;
  expiresAt: number;
}
