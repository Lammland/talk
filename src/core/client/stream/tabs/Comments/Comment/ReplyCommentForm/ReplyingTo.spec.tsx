import React from "react";
import { createRenderer } from "react-test-renderer/shallow";

import { PropTypesOf } from "coral-framework/types";

import ReplyingTo from "./ReplyingTo";

it("renders correctly", () => {
  const props: PropTypesOf<typeof ReplyingTo> = {
    username: "ParentAuthor",
  };
  const renderer = createRenderer();
  renderer.render(<ReplyingTo {...props} />);
  expect(renderer.getRenderOutput()).toMatchSnapshot();
});
