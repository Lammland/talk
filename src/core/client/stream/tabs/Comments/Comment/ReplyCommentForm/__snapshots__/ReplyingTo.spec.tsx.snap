// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<ForwardRef(forwardRef)
  alignItems="center"
  className="ReplyingTo-root coral coral-createReplyComment-replyTo"
>
  <ForwardRef(forwardRef)>
    reply
  </ForwardRef(forwardRef)>
   
  <Localized
    Username={<Username />}
    id="comments-replyingTo"
  >
    <span
      className="ReplyingTo-text coral coral-createReplyComment-replyToText"
    >
      Replying to &lt;Username&gt;&lt;/Username&gt;
    </span>
  </Localized>
</ForwardRef(forwardRef)>
`;
